City-of-Helsinki/drupal-repository
# UHF-9884: Queue package rebuild command, test coverage

## How to install
- `git checkout UHF-9884`
- `composer install`
- `docker compose build`
- `docker compose up -d`
- `docker compose exec app sh`
- `cd /app`

## How to test
- Make sure `dist/` folder does not exist
- Make sure the queue consumer is started (run `ps aux|grep consume` inside the container)
- Run `php console.php queue:package drupal/helfi_api_base` and make sure `dist/p2/drupal/helfi_api_api*` files are written
- Run `php console.php queue:package all` and make sure the entire index is re-populated (this can take up to 20 minutes)
  - You can check the progress with `ps aux | grep satis`
- See [Test webhooks locally](https://github.com/City-of-Helsinki/drupal-repository/blob/UHF-9884/documentation/development.md#test-webhooks-locally) documentation and make sure you can trigger the rebuild command via Webhook
   - Curl should respond with `200` code
   - Check the app logs with `docker compose logs app` to verify the command was run

## Other PRs
- https://github.com/City-of-Helsinki/drupal-docker-images/pull/26
